src/client/inputhandler.h

```diff
@@ -269,72 +293,97 @@ class RandomInputHandler : public InputHandler
 	{
 		leftdown = false;
 		rightdown = false;
+		middledown = false;
 		leftclicked = false;
 		rightclicked = false;
+		middleclicked = false;
 		leftreleased = false;
 		rightreleased = false;
+		middlereleased = false;
 		keydown.clear();
 	}
 	virtual bool isKeyDown(const KeyPress &keyCode)
 	{
+		if (keyCode == getKeySetting("keymap_dig")) {
+			return leftclicked;
+		}
+		if (keyCode == getKeySetting("keymap_place")) {
```

Comment:
else if
(I don't see how keyCode can have more than one value at once. I guess getKeySetting("keymap_dig") and getKeySetting("keymap_place") might be able to map to the same keyCode but surely that would be a bug)